### DISCUSSION TITLE
Why does block size is limited at 16Mb ?

### DISCUSSION THREAD

**Discussion by chambo-e** [juicedata/juicefs#738] [Q&A] [answered]
Hello, 

Just saw that blockSize is limited at 16MB while trying to set one at 32MB https://github.com/juicedata/juicefs/blob/3190a02517f26ce7fd2cd0e0bcbed8efb301ba7d/cmd/format.go#L39

I wasn't able to find a mention of this limit in the docs, was wondering if there was any ? 

Thanks :)

**Comment by davies** [comment, marked as answer]
Block is the unit of PUT/GET and also cached in disk. For bigger block size, it will consume more memory and easy to fail, and have not   done much tests with them.

Why do you prefer bigger block size?

**Reply by chambo-e** [reply]
I use an S3 provider which recommend to keep < 500K objects in a bucket, otherwise performance might suffer

Having bigger blocks means I could stay below 500k :)

**Reply by xiaogaozi** [reply]
Maybe you should try `--shards` option in [`juicefs format`](https://github.com/juicedata/juicefs/blob/main/docs/en/command_reference.md#juicefs-format) command.

**Reply by chambo-e** [reply]
Ooh, I missed that option, thanks a lot, works like a charm !

**Reply by jheredianet** [reply]
Please could you give an example of using this?
Thanks in advance.
